core.go

```diff
@@ -286,18 +327,19 @@ func (l *LevelEnabler) Enabled(lvl zapcore.Level) bool {
 // follow same logic with sentry-go to filter unnecessary frames
 // ref:
 // https://github.com/getsentry/sentry-go/blob/362a80dcc41f9ad11c8df556104db3efa27a419e/stacktrace.go#L256-L280
-func filterFrames(frames []sentry.Frame) []sentry.Frame {
+func filterFrames(frames []sentry.Frame, filters FrameFilters) []sentry.Frame {
 	if len(frames) == 0 {
 		return nil
 	}
 
-	for i := range frames {
-		// Skip zapsentry and zap internal frames, except for frames in _test packages (for
-		// testing).
-		if (strings.HasPrefix(frames[i].Module, "github.com/TheZeroSlave/zapsentry") ||
-			strings.HasPrefix(frames[i].Function, "go.uber.org/zap")) &&
-			!strings.HasSuffix(frames[i].Module, "_test") {
-			return frames[0:i]
+	for idx := 0; idx < len(frames); {
+		if filters.AnyMatch(frames[idx]) {
+			if idx < len(frames)-1 {
+				copy(frames[idx:], frames[idx+1:])
+			}
+			frames = frames[:len(frames)-1]
+		} else {
```

Comment:
Please refactor this to "return early" (replace `else` with `continue` in the other branch). And also, adding empty lines after the end of a block improves readability IMO, but that is just my opinion.